lib/kerberos.js

```diff
@@ -36,6 +38,93 @@ function validateParameter(parameter, spec) {
   }
 }
 
+/// allows api methods to be called with a callback or a promise
+function promisify(fn, paramDefs) {
+  return function() {
+    const args = Array.prototype.slice.call(arguments);
+    const params = [];
+    for (let i = 0; i < paramDefs.length; ++i) {
+      const def = paramDefs[i];
+      let arg = args[i];
+
+      if (def.default && arg == null) arg = def.default;
+      if (def.type === 'object' && def.default != null) {
+        arg = Object.assign({}, def.default, arg);
+      }
+
+      // special case to allow `options` to be optional
+      if (def.name === 'options' && (typeof arg === 'function' || arg == null)) {
+        arg = {};
+      }
+
+      validateParameter(arg, paramDefs[i]);
+      params.push(arg);
+    }
+
+    const callback = arguments[arguments.length - 1];
+    if (typeof callback !== 'function') {
+      return new Promise((resolve, reject) => {
+        params.push((err, response) => {
+          if (err) return reject(err);
+          resolve(response);
+        });
+
+        fn.apply(this, params);
+      });
+    }
+
+    params.push(callback);
+    fn.apply(this, params);
+  };
+}
+
+/**
+ * Processes a single GSSAPI client-side step using the supplied server data.
+ *
+ * @param {string} challenge A string containing the base64-encoded server data (which may be empty for the first step)
+ * @param {function} callback Returns a result code, or an error if one was
```

Comment:
Maybe rephrase "or an error if one was" as "or an error if one occurred"

Reply:
yeah I think documentation is a bit incomplete, that's the next step 😄